bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/ApiBridgeHandler.java

```diff
@@ -0,0 +1,227 @@
+
+        super.dispose();
+    }
+
+    /**
+     * Creates a new {@link LinkyAuthServlet}.
+     *
+     * @return the newly created servlet
+     * @throws IOException thrown when an HTML template could not be read
+     */
+    private HttpServlet createServlet() throws LinkyException {
+        return new LinkyAuthServlet(this);
+    }
+
+    public String authorize(String redirectUri, String reqState, String reqCode) throws LinkyException {
+        // Will work only in case of direct oAuth2 authentification to enedis
+        // this is not the case in v1 as we go trough MyElectricalData
+
+        try {
+            logger.debug("Make call to Enedis to get access token.");
+            final AccessTokenResponse credentials = oAuthService
+                    .getAccessTokenByClientCredentials(LinkyBindingConstants.LINKY_SCOPES);
+
+            String accessToken = credentials.getAccessToken();
+
+            logger.debug("Acces token: {}", accessToken);
+            return accessToken;
+        } catch (RuntimeException | OAuthException | IOException e) {
+            throw new LinkyException("Error during oAuth authorize :" + e.getMessage(), e);
+        } catch (final OAuthResponseException e) {
+            throw new LinkyException("Error during oAuth authorize :" + e.getMessage(), e);
+        }
+    }
+
+    public boolean isAuthorized() {
+        final AccessTokenResponse accessTokenResponse = getAccessTokenResponse();
+
+        return accessTokenResponse != null && accessTokenResponse.getAccessToken() != null
+                && accessTokenResponse.getRefreshToken() != null;
+    }
+
+    protected @Nullable AccessTokenResponse getAccessTokenByClientCredentials() {
+        try {
+            return oAuthService.getAccessTokenByClientCredentials(LinkyBindingConstants.LINKY_SCOPES);
+        } catch (OAuthException | IOException | OAuthResponseException | RuntimeException e) {
+            logger.debug("Exception checking authorization: ", e);
+            return null;
+        }
+    }
+
+    protected @Nullable AccessTokenResponse getAccessTokenResponse() {
+        try {
+            return oAuthService.getAccessTokenResponse();
+        } catch (OAuthException | IOException | OAuthResponseException | RuntimeException e) {
+            logger.debug("Exception checking authorization: ", e);
+            return null;
+        }
+    }
+
+    public String formatAuthorizationUrl(String redirectUri) {
+        try {
+            String uri = this.oAuthService.getAuthorizationUrl(redirectUri, LinkyBindingConstants.LINKY_SCOPES,
+                    LinkyBindingConstants.BINDING_ID);
+            return uri;
+        } catch (final OAuthException e) {
+            logger.debug("Error constructing AuthorizationUrl: ", e);
+            return "";
+        }
+    }
+
+    public List<String> getAllPrmId() {
+        List<String> result = new ArrayList<>();
+
+        Collection<Thing> col = this.thingRegistry.getAll();
```

Comment:
Instead of searching all things and collect their configuration information, things should provide their prmId to the api bridge.

Reply:
Ok, i reverse the logic so each LinkyHandler register to the bridge.
Fix in next commit